components/oauth/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuthApplicationMgtListener.java

```diff
@@ -0,0 +1,147 @@
+        removeClientSecret(serviceProvider);
+        return true;
+    }
+
+    public boolean doPostGetApplication(ServiceProvider serviceProvider, String serviceProviderName, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (serviceProvider == null) {
+            return true; // if service provider is not present no need to add this information
+        }
+
+        try {
+            addClientSecret(serviceProvider, serviceProviderName, tenantDomain);
+        } catch (IdentityOAuthAdminException e) {
+            throw new IdentityApplicationManagementException("Injecting client secret failed.", e);
+        }
+
+        return true;
+    }
+
+    public boolean doPostGetApplicationByClientId(ServiceProvider serviceProvider, String clientId, String clientType,
+                                                  String tenantDomain) throws IdentityApplicationManagementException {
+        if (serviceProvider == null) {
+            return true; // if service provider is not present no need to add this information
+        }
+
+        try {
+            addClientSecret(serviceProvider, clientType, tenantDomain);
+        } catch (IdentityOAuthAdminException e) {
+            throw new IdentityApplicationManagementException("Injecting client secret failed.", e);
+        }
+
+        return true;
+    }
+
+    private void removeClientSecret(ServiceProvider serviceProvider) {
+        InboundAuthenticationConfig inboundAuthenticationConfig = serviceProvider.getInboundAuthenticationConfig();
+        if (inboundAuthenticationConfig != null) {
+            InboundAuthenticationRequestConfig[] inboundRequestConfigs = inboundAuthenticationConfig.
+                    getInboundAuthenticationRequestConfigs();
+            if (inboundRequestConfigs != null) {
+                for (InboundAuthenticationRequestConfig inboundRequestConfig : inboundRequestConfigs) {
+                    if (inboundRequestConfig.getInboundAuthType().equals(OAUTH2)) {
+                        Property[] props = inboundRequestConfig.getProperties();
+                        for (Property prop : props) {
+                            if (prop.getName().equalsIgnoreCase(OAUTH2_CONSUMER_SECRET)) {
+                                props = (Property[]) ArrayUtils.removeElement(props, prop);
+                                inboundRequestConfig.setProperties(props);
+                                continue;   //we are interested only on this property
+                            } else {
+                                //ignore
+                            }
+                        }
+                        continue;// we are interested only on oauth2 config. Only one will be present.
+                    } else {
+                        //ignore
+                    }
+                }
+            } else {
+
+            }
+        } else {
+            //nothing to do
+        }
+    }
```

Comment:
Learnt in early code reviews to keep these as a best pratice, so that if later some other developer is to put some logic in, they will explicitly look for else logic placement. Otherwise there is possibility someone placing a logic that require to be our of else within an invisible else.. Should I still remove it?